src/coreclr/jit/gentree.cpp

```diff
@@ -7900,6 +8007,20 @@ GenTree* Compiler::gtClone(GenTree* tree, bool complexOK)
             copy = gtNewLconNode(tree->AsLngCon()->gtLconVal);
             break;
 
+        case GT_CNS_DBL:
+        {
+            copy = gtNewDconNode(tree->AsDblCon()->gtDconVal, tree->TypeGet());
```

Comment:
Is this code preserving existing behavior or extending it?

Note that `gtClone` is (by design) very conservative in what it clones, e. g. it will prefer not to to clone if the `tree` is a load.

Reply:
Its preserving existing behavior, it just explicitly passes the type to the create call rather than manually assigning it afterwards.

Reply:
Was your comment mean to apply to `gtCloneExpr`? For that function there are indeed no changes of note.

For `gtClone`, I do not see it cloning `CNS_DBL` (or `CNS_VEC` equivalents) in [`main`](https://github.com/dotnet/runtime/blob/6010dc0604a9f2708dba8d21f57d97b613dfcabe/src/coreclr/jit/gentree.cpp#L7875).